addons/l10n_in/tests/test_l10n_in_fiscal_position.py

```diff
@@ -132,3 +132,63 @@ def test_l10n_in_fiscal_in_bill_to_ship_to(self):
             out_invoice.fiscal_position_id,
             self.env['account.chart.template'].ref('fiscal_position_in_export_sez_in')
         )
+
+    def test_l10n_in_fiscal_in_vendor_bills(self):
+        self.env.company = self.default_company
+        company_res_state = self.default_company.state_id
+        self.partner_a.write({
+            'state_id': company_res_state.id,
+            'country_id': company_res_state.country_id.id,
+        })
+        other_state = self.env['res.country.state'].search([
+            ('id', '!=', company_res_state.id), ('country_id', '=', company_res_state.country_id.id)],
+            limit=1
+        )
+        # Intra State
+        vendor_bill_1 = self._assert_invoice_fiscal_position(
+            fpos_ref='fiscal_position_in_intra_state',
+            partner=self.partner_a,
+            taxes=self.sgst_sale_18,
+            move_type='in_invoice',
+            post=False,
+        )
+        # Inter State
+        vendor_bill_1.write({
+            'l10n_in_state_id': other_state.id,
+        })
+        self.assertEqual(
+            vendor_bill_1.fiscal_position_id,
+            self.env['account.chart.template'].ref('fiscal_position_in_inter_state')
+        )
+        self.partner_a.write({
+            'state_id': other_state.id,
+        })
+        vendor_bill_2 = self._assert_invoice_fiscal_position(
+            fpos_ref='fiscal_position_in_inter_state',
+            partner=self.partner_a,
+            taxes=self.sgst_sale_18,
+            move_type='in_invoice',
+            post=False,
+        )
+        vendor_bill_2.write({
+            'l10n_in_state_id': other_state.id,
+        })
+        self.assertEqual(
+            vendor_bill_2.fiscal_position_id,
+            self.env['account.chart.template'].ref('fiscal_position_in_intra_state')
```

Comment:
That does not make any sense.

Reply:
The test case? Or the way of working? Because this how we are expecting it to work correctly